crates/pico/src/source.rs

```diff
@@ -9,6 +9,10 @@ pub trait Source {
     fn get_key(&self) -> Key;
 }
 
+/// A marker trait that is implemented by the #[derive(Singleton)]
+/// proc-macro.
+pub trait Singleton {}
```

Comment:
What if we do:
```rs
pub trait Singleton: Source {
    fn get_key() -> Key;
}
```
and use basically the same macro as for `get_key(&self)` in `Source`?
Then we can:
- use `db.set` for both source and singleton
- use singleton type to get keys like `<T as Singleton>::get_key()` for generics of `SchemaFile::get_key()` for implementations.